### ISSUE TITLE
Testing framework

### ISSUE THREAD

**Issue by sebiwi** [opsmop/opsmop#58]
Hi,

I was wondering how testing should be addressed when using Opsmop.
IMHO, one of the greatest advantages of using Python insteal of YAML
is the possibility of unit testing your code in a simple fashion.

I would love to write my Roles and Policies using Test Driven Development,
for example, as it would allow me to have amazingly fast feedback on the
code I'm writing without having any side effects, or without having to provision
infrastructure in order to do so. I'd love to say in Python all the way.

This is not the same as `validate` mode. I want to actually test the logic and intention
of my code.  

Basically, I'd love to be able to do something like this: https://sebiwi.github.io/blog/the-wizard-1/
but without having to deploy ephemeral infrastructure in order to test the code. 

Thanks in advance, and thank you for Opsmop! (This is great! \o/)

**Comment by mpdehaan** [comment]
Hi @sebiwi,

First off, great question!

This is more of a question than a ticket, so can you please ask this over at http://talk.msphere.io?

I'm wanting to encourage the forum for topics like this - not only is it more browseable, it helps keep my ticket s
ystem organized.

There was a bit of an existing idea that I can link to when you start the new thread, and we can see what ideas come up.

Thank you!
